README.md

```diff
@@ -1,31 +1,118 @@
+
+```
+python sample_structure.py --bulk_db bulk_db_flat.pkl \
+  --adsorbate_db adsorbate_db.pkl \
+  --output_dir outputs/  \
+  --enumerate_all_structures \
+  --adsorbate_index 10 --bulk_indices 20
 ```
-python sample_structure.py --bulk_db bulk_database.pkl --adsorbate_db adsorbate_database.pkl --output_dir outputs/ --enumerate_all_structures --adsorbate_index 10 --bulk_indices 20
+
+1. Generate one specified adsorbate, one or more specified bulks, one specified surface, and all possible configs. This is the same as #2 except only one surface is selected, which can be used to help parallelize #2. The following example generates files for all adsorbate placements of adsorbate 10 on surface 0 from bulk 20, resulting in files in `outputs/10_20_0/surface/`, `outputs/10_20_0/adslab0/`, `outputs/10_20_0/adslab1/`, and so on for all the adsorbate+surface configs.
+
 ```
+python sample_structure.py --bulk_db bulk_db_flat.pkl \
+  --adsorbate_db adsorbate_db.pkl --output_dir outputs/ 
+  --enumerate_all_structures --adsorbate_index 10 
+  --bulk_indices 20 --surface_index 0
+```
+
+For any of the above, `--precomputed_structures dir/`  can be added to use the precomputed surfaces rather than calculating all possible surfaces of a given bulk from scratch. Adding `--verbose` will print out additional info.
+
+
+## Generating inputs for OC20
+
+### Background
+
+We employed the workflow as described above to create the OC20 dataset. We considered bulk materials with up to 3 elements. A total of ~15k bulk structures and 82 adsorbates were considered for sampling. The resulting decision tree was huge since the branching factor at every stage in the workflow is reasonably big. For instance, sampling a surface from a chosen bulk requires considering at most 125 different miller-index calculations. Moreover, placing an adsorbate on a given surface requires enumerating ~30 sites on an average.
+
+We generated around 2.5M adsorbate+catalyst inputs which were then deduplicated before running DFT calculations.  Each input system was generated on a single CPU core with a timeout of 200 minutes.
+
+
+### Databases for bulk, adsorbate and precomputed surfaces 
+
+**Bulks**
+Bulk database that is a dict where keys are the number of elements (1-3), and values are lists of bulks, for use case #1 above where we randomly sample a bulk given the number of elements:
+
+* https://dl.fbaipublicfiles.com/opencatalystproject/data/input_generation/bulk_db_nelems_2020may12.pkl (MD5 checksum: `b87281d01a2006586fa719a1d2ee2682`)
+
+Flat bulk database (all 11k bulks in one list), for use cases #2 and #3 above, where we want to retrieve bulks by their index:
+
+* https://dl.fbaipublicfiles.com/opencatalystproject/data/input_generation/bulk_db_flat_2020may12.pkl (MD5 checksum: `aaf192104fb51be84fdcf9fd7d3aae49`)
+
+
+**Adsorbates**
+https://dl.fbaipublicfiles.com/opencatalystproject/data/input_generation/adsorbate_db_2020may12.pkl (MD5 checksum: `6c553a7c2009e11612c247ffb813fd74`)
```

Comment:
Just remembered this - I updated one of the adsorbates in the adsorbate.db and adsorbate.pkl after vv1k enumeration and it's not reflected here. It's the 28th one. Instead of 
**(Atoms(symbols='C2H2O', pbc=False), '*CH*COH', array([0, 1]))**
It should be **(Atoms(symbols='C2H2O', pbc=False), 'CH*COH', array([0]))**

Reply:
Since this was updated after the OC20 dataset, I am thinking of leaving the current db as is (for consistency with OC20). If those changes are important, perhaps we should also upload a new version?